server.go

```diff
@@ -179,6 +180,8 @@ type server struct {
 	txIndex         *indexers.TxIndex
 	addrIndex       *indexers.AddrIndex
 	existsAddrIndex *indexers.ExistsAddrIndex
+
+	pendingInvs map[wire.InvVect]interface{}
```

Comment:
Please don't move this out of the rebroadcast handler.  It's no longer concurrent safe moving it out here.  You've introduced races by doing this.